source/MQTTFileDownloader.c

```diff
@@ -81,13 +97,37 @@ static MQTTFileDownloaderStatus_t handleCborMessage( uint8_t * decodedData,
  * @param[in] messageLength Length of the MQTT message received.
  * @param[out] decodedData Buffer to place the decoded data in.
  * @param[out] decodedDataLength Length of decoded data.
+ * @param[out] fileId ID of the file to which the data block belongs.
+ * @param[out] blockId ID of the received block.
+ * @param[out] blockSize Size of the receive block in bytes.
  *
  * @return uint8_t returns appropriate MQTT File Downloader Status.
  */
-static MQTTFileDownloaderStatus_t handleJsonMessage( uint8_t * decodedData,
-                                                     size_t * decodedDataLength,
-                                                     uint8_t * message,
-                                                     size_t messageLength );
+static MQTTFileDownloaderStatus_t handleJsonMessage( uint8_t * message,
+                                                     size_t messageLength,
+                                                     int32_t * fileId,
+                                                     int32_t * blockId,
+                                                     int32_t * blockSize,
+                                                     uint8_t * decodedData,
+                                                     size_t * decodedDataLength );
+
+/**
+ * @brief Handles and decodes the received message in JSON format.
+ *
+ * @param[in] str String which has the ASCII representation of the
+ *            number.
+ * @param[in] len Length of the string.
+ * @param[out] num Pointer to an int32_t variable which will be filled
+ *            with the parsed value. If the return value of the function
+ *            call is -1, then the value present in num is invalid.
```

Comment:
I thought a bool could only be 0 or 1 in C, can it return -1?

Reply:
Good catch. Fixed with https://github.com/aws/aws-iot-core-mqtt-file-streams-embedded-c/pull/23/commits/f1f6bdba40afd6af472fc0412d5ea99a555623f7